src/pymor/reductors/aaa.py

```diff
@@ -0,0 +1,421 @@
+                v = 1j * rng.normal(scale=np.sqrt(2)/2, size=(dim_input,)) \
+                    + rng.normal(scale=np.sqrt(2)/2, size=(dim_input,))
+            else:
+                w = rng.normal(size=(dim_output,))
+                v = rng.normal(size=(dim_input,))
+            w /= np.linalg.norm(w)
+            v /= np.linalg.norm(v)
+            samples_T = samples @ v @ w
+            samples_orig = samples
+            samples = samples_T
+        else:
+            dim_input = 1
+            dim_output = 1
+
+        # initialize data partitions, error, max iterations
+        err = np.inf
+        if itpl_part is None:
+            self.itpl_part = [[] for _ in range(num_vars)]
+        else:
+            assert len(itpl_part) == num_vars
+            self.itpl_part = itpl_part
+        if max_itpl is None:
+            max_itpl = [len(s)-1 for s in svs]
+
+        assert len(max_itpl) == len(svs)
+
+        # start iteration with constant function
+        bary_func = np.vectorize(lambda *args: np.mean(samples))
+
+        # iteration counter
+        j = 0
+
+        while any(len(i) < mi for i, mi in zip(self.itpl_part, max_itpl)):
+
+            # compute approximation error over entire sampled data set
+            grid = np.meshgrid(*svs, indexing='ij')
+            err_mat = np.abs(bary_func(*grid) - samples)
+
+            # set errors to zero such that new interpolation points are consistent with max_itpl
+            zero_idx = []
+            for i in range(num_vars):
+                if len(self.itpl_part[i]) >= max_itpl[i]:
+                    zero_idx.append(list(range(samples.shape[i])))
+                else:
+                    zero_idx.append(self.itpl_part[i])
+            err_mat[np.ix_(*zero_idx)] = 0
+            err = np.max(err_mat)
+
+            j += 1
+            self.logger.info(f'Relative error at step {j}: {err/max_samples:.5e}, '
+                             f'number of interpolation points {[len(ip) for ip in self.itpl_part]}')
+
+            # stopping criterion based on relative approximation error
+            if err <= rel_tol:
+                break
+
+            greedy_idx = np.unravel_index(err_mat.argmax(), err_mat.shape)
+            for i in range(num_vars):
+                if greedy_idx[i] not in self.itpl_part[i] and len(self.itpl_part[i]) < max_itpl[i]:
+                    self.itpl_part[i].append(greedy_idx[i])
+
+                    # perform double interpolation step to enforce real state-space representation
+                    if i == 0 and self.conjugate and np.imag(svs[i][greedy_idx[i]]) != 0:
+                        conj_sample = np.conj(svs[i][greedy_idx[i]])
+                        conj_idx = np.where(svs[0] == conj_sample)[0]
+                        self.itpl_part[i].append(conj_idx[0])
+
+            # solve LS problem
+            L = full_nd_loewner(samples, svs, self.itpl_part)
+
+            _, S, V = spla.svd(L, lapack_driver='gesvd')
```

Comment:
Is [`full_matrices==True`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.svd.html) necessary here?